testlapack/dlaqr23.go

```diff
@@ -27,20 +27,23 @@ type dlaqr2Test struct {
 	evWant []complex128 // Optional slice with known eigenvalues.
 }
 
-func Dlaqr2Test(t *testing.T, impl Dlaqr2er) {
+func Dlaqr23Test(t *testing.T, impl Dlaqr23er) {
 	rnd := rand.New(rand.NewSource(1))
 
 	for _, wantt := range []bool{true, false} {
 		for _, wantz := range []bool{true, false} {
-			for _, n := range []int{1, 2, 3, 4, 5, 6, 10, 18, 31, 53} {
-				for _, extra := range []int{0, 1, 11} {
-					for cas := 0; cas < 100; cas++ {
-						ktop := rnd.Intn(n)
-						kbot := rnd.Intn(n)
-						if ktop > kbot {
-							ktop, kbot = kbot, ktop
+			for _, n := range []int{1, 2, 3, 4, 5, 6, 10, 18, 31, 100} {
+				for _, extra := range []int{0, 11} {
+					for cas := 0; cas < 30; cas++ {
+						var nw int
+						if nw <= 75 {
+							nw = rnd.Intn(n) + 1
+						} else {
+							nw = 76 + rnd.Intn(n-75)
```

Comment:
I changed the logic here to make sure that for sufficiently large matrices we generate large enough window to trigger the Dlaqr4 path. For small matrices any window size works because they will always use Dlahqr inside Dlaqr23.